Question-2/contactList/ContactsViewDelegate.h

```diff
@@ -0,0 +1,23 @@
+//
+//  ContactsViewDelegate.h
+//  contactList
+//
+//  Created by Bharath Reddy on 27/11/22.
+//
+
+#import <Foundation/Foundation.h>
+#import "UIKit/UIKit.h"
+#import "DetailsViewController.h"
+#import "ContactDataModel.h"
+NS_ASSUME_NONNULL_BEGIN
+@protocol  showingDetailsDelegate 
+
+-(void)showDetails:(DetailsViewController *)detailsViewController;
+
+@end
+@interface ContactsViewDelegate :NSObject
+@property (nonatomic,weak) id<showingDetailsDelegate> delegate;
+- (instancetype)initWithContacts:(NSArray<ContactDataModel *> *)contacts;
+@end
```

Comment:
NIT: We add new line after declaring all the properties and inits.
 
This should be like -
```
@interface ContactsViewDelegate : NSObject

@property (nonatomic,weak) id<showingDetailsDelegate> delegate;

- (instancetype)initWithContacts:(NSArray<ContactDataModel *> *)contacts;

@end
```

if you have multiple properties nad inits, then this should be like -
```
@interface ContactsViewDelegate : NSObject

@property (nonatomic,weak) id property1;
@property (nonatomic,weak) id property2;
@property (nonatomic,weak) id property3;

- (instancetype)initWithContacts1:(NSArray<ContactDataModel *> *)contacts1;
- (instancetype)initWithContacts2:(NSArray<ContactDataModel *> *)contacts2;

@end
```